### PR TITLE
Assert on evaluation results after classification analysis in tests.

### PR DESCRIPTION
**Pull request by przemekwitek**
Assert that the results of classification analysis can be evaluated using _evaluate API.
Remove testSingleNumericFeatureAndMixedTrainingAndNonTrainingRows_TopClassesRequested as redundant.

Relates https://github.com/elastic/elasticsearch/issues/46735

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java [hunk 1]
```diff
@@ -377,4 +343,26 @@ private static void indexData(String sourceIndex, int numTrainingRows, int numNo
         // Assert that the top classes are listed in the order of decreasing probabilities.
         assertThat(Ordering.natural().reverse().isOrdered(classProbabilities), is(true));
     }
+
+    private void assertEvaluation(String dependentVariable, List<String> dependentVariableValues, String predictedClassField) {
+        EvaluateDataFrameAction.Response evaluateDataFrameResponse =
+            evaluateDataFrame(
+                destIndex,
+                new org.elasticsearch.xpack.core.ml.dataframe.evaluation.classification.Classification(
```
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java]: java needs import alias :)

**Review by benwtrent — APPROVED** [review]

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java]: Fully agree...
Another option would be to rename the class to `ClassificationEvaluation` (similarly with `Regression`) but this is outside the scope of this PR

**Comment by przemekwitek** [comment]
run elasticsearch-ci/packaging-sample-matrix

**Pull request merged**